### DISCUSSION TITLE
Custom React Templates

### DISCUSSION THREAD

**Discussion by lr001dev** [nrwl/nx#4843] [Feature Requests]
Does NX currently support creating a custom React app from the generator? I ideally I'd like to add some custom boilerplate when generating new react apps inside the workspace. Is there any info on how to accomplish this? CRA has something like this a reference.

**Comment by FrozenPandaz** [comment]
Absolutely! :smile:

We realize that every organization has their own standards and one of the ways those custom standards can be enforced is by providing code generation to introduce the standards from the beginning.

**Workspace Generators** are custom code generation that you can write for your workspace to do stuff like adding custom boilerplate when generating new apps. Read the full guide here: https://nx.dev/latest/react/generators/workspace-generators

**Reply by furai** [reply]
Seems that this feature got replaced with https://nx.dev/extending-nx/recipes/local-generators
